### PR TITLE
Folders: Fix creating/updating a folder whose title has leading and trailing spaces

### PR DESCRIPTION
**Pull request by papagian**
**What is this feature?**

Temporarily folders are store in both `dashboard` and `folder` table. This fixes an inconsistency in two tables that used to occur if the folder title used to contain leading and trailing spaces because if the `folder` table was stored as is, while in the `dashboard` table was store trimmed.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by yuri-tceretian** [comment]
Since this is already in the wild, I think we need a migration too.

File: pkg/services/folder/folderimpl/folder_test.go [hunk 1]
```diff
@@ -732,6 +732,70 @@ func TestNestedFolderServiceFeatureToggle(t *testing.T) {
...
+func TestFolderServiceDualWrite(t *testing.T) {
+	g := guardian.New
+	guardian.MockDashboardGuardian(&guardian.FakeDashboardGuardian{CanSaveValue: true})
+	t.Cleanup(func() {
+		guardian.New = g
+	})
+
+	db := sqlstore.InitTestDB(t)
+	cfg := setting.NewCfg()
+	features := featuremgmt.WithFeatures()
```
Comment (diegommm) [inline pkg/services/folder/folderimpl/folder_test.go:744]: [question] [non-blocker] Do we need `ProvideStore` to receive an empty `*FeatureManagement` or could we create `features` like this and reuse it when creating `Service`? 
```suggestion
	features := featuremgmt.WithFeatures(featuremgmt.FlagNestedFolders)
```

**Review by diegommm — APPROVED** [review]

File: pkg/services/folder/folderimpl/folder_test.go (hunk 1 above)
Comment (papagian) [reply pkg/services/folder/folderimpl/folder_test.go:744]: actually `features` is redundant (is not used inside the nested folder store) so I have removed it
but to answer your question: yes, we can create the features like this and pass the same to the service.

**Comment by papagian** [comment]
> Since this is already in the wild, I think we need a migration too.

this will be addressed by #80972

**Pull request merged**